eclipse-theia/theia
# Implement Disassembly View and InstructionBreakpoints

#### What it does


> Leaving this as a draft for now, as there are some details I need to check to make sure instruction breakpoints are working correctly.

Addresses parts of #8136 by implementing InstructionBreakpoints and a disassembly view to and the return of the disassemble request.

This PR copies and adapts VSCode's implementation of the Disassembly View for Theia.

#### How to test


0. In general: use any debugger that supports the disassemble request, and confirm that the behavior in Theia is consistent with the behavior in VSCode.
1. For example...
2. Install the [CDT-GDB VSCode plugin](https://open-vsx.org/extension/eclipse-cdt/cdt-gdb-vscode)
3. Open [this example workspace](https://github.com/eclipse-theia/theia-cpp-extensions/tree/master/examples/cpp-debug-workspace) and execute the build task.
4. Set a breakpoint in `a.cpp` and start debugging.
5. When you hit a breakpoint, you should be able to open the context menu in the stack view and find the 'Open Disassembly View' command - it should also be available in the command palette.
6. Disassembly should load.
7. Scrolling should load more lines.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
